src/nspawn/nspawn.c

```diff
@@ -3893,7 +3985,9 @@ int main(int argc, char *argv[]) {
         }
 
         if (remove_tmprootdir) {
-                if (rmdir(tmprootdir) < 0)
+                if (rmdir (tmprootdir_root) < 0 ||
+                    rmdir (tmprootdir_tmp) < 0 ||
```

Comment:
thetmpdir doesn't have to stay around so long, does it? you can remove it again after you switched things around immediately, no?